### DISCUSSION TITLE
Read a YAML file to a string

### DISCUSSION THREAD

**Discussion by hjiajing** [mikefarah/yq#1336] [Q&A] [answered]
Can we use `yq` to read a YAML file to a string?

e.g.
The file is 
```yaml
test:
  a: b
```
and what I want is `test: \n. a: b\n`

**Comment by mikefarah** [comment]
Not entirely sure what you want here - as the file is already a string. 

I think you probably need to use the encoder operator detailed here: https://mikefarah.gitbook.io/yq/operators/encode-decode

**Comment by julianbarg** [comment, marked as answer]
Since yq's output is already a string, you can do any kind of conversion you want on that output. Would this work for you?

```bash
yq '.' file.yaml | awk '{printf "%s\\n", $0}'
```

**Reply by gswallow** [reply]
Why does this have to be done with awk?

**Comment by mikefarah** [comment]
Oh I see, you can get the same result using the 'sub' command to replace newlines with \n, no need for awk.
________________________________

**Reply by hjiajing** [reply]
Thanks for the reply, I used `sed` to replace the "\n" and then it works well.
